src/components/transactions/TxSummary/PendingActions.tsx

```diff
@@ -0,0 +1,28 @@
+import { Box } from '@mui/material'
+import { getTransactionDetails, type TransactionSummary } from '@safe-global/safe-gateway-typescript-sdk'
+import TxStatusLabel from '@/components/transactions/TxStatusLabel'
+import { SpeedUpMonitor } from '@/features/speedup/components/SpeedUpMonitor'
+import useAsync from '@/hooks/useAsync'
+import { useAppSelector } from '@/store'
+import { selectPendingTxById } from '@/store/pendingTxsSlice'
+
+export const PendingActions = ({ tx }: { tx: TransactionSummary }) => {
+  const pendingTx = useAppSelector((state) => selectPendingTxById(state, tx.id))
+
+  const [txDetails, txDetailsLoading, txDetailError] = useAsync(() => {
+    if (!pendingTx) return Promise.resolve(undefined)
```

Comment:
This block will be shown for every tx in the queue.
Pending txs are only txs that are currently executing, processing or indexing though.
For the others it will be undefined. Sadly the types do not reflect that as we use a `Record<string, PendingTx>` here.